website/docs/index.md

```diff
@@ -53,6 +53,41 @@ ln -s -f ../../scripts/pre-commit .git/hooks/pre-commit
 ```
 
 
+## How to Create a Good UNIX Interface
+
+One of our goals with this project is learning how to write programs and, more importantly,
+the importance of not writing code vs. writing code. 
+
+Here are some 
+- [good rules for writing good unix programs](https://lasr.cs.ucla.edu/ficus-members/geoff/interfaces.html)
+
+A good example can be seen in notions of progress bars. The idea of progress bars has come up twice now, from 
+two different contributors, once in cp, and once in dd.
+
+What's wrong with adding a progress bar to a program? In short, everything!
+Where does the output go? stdout? Then you can not add it to some programs because their output is to stdout. 
+So do you add it to stderr? But it's not an error, is it? It's informational. 
```

Comment:
According to posix standard, stderr is for `for writing diagnostic output`, not necessarily errors.
This is the same meaning many programming languages use by default:

- Go writes `log.Print` to stderr
- C++ writes `clog` to stderr